### PR TITLE
[Merged by Bors] - Dataplane protocol support for wasm

### LINKED ISSUE #1100: Add wasm32 support to dataplane-protocol crate
The file feature flag that doesn't work with wasm32. Let's add the appropriate `#[cfg]` stuff to the `file` `cfg`s to work with wasm32.

This is a subtask of #989.

### PR DESCRIPTION
**Pull request by simlay**
Closes #1100 and I think #1099.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -581,4 +581,33 @@ jobs:
...
+          pkill -f "minikube tunnel" || true
+
+  wasm:
+    name: Check wasm support (${{ matrix.wasm-crate }})
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        rust: [stable]
+        wasm-crate: [spu-schema, sc-schema, dataplane-protocol, types, protocol]
```
Comment (simlay) [inline .github/workflows/ci.yml:593]: I've got mixed feelings about doing it this way.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -581,4 +581,33 @@ jobs:
...
+    env:
+      RUST_BACKTRACE: full
+      RUSTC_WRAPPER: sccache
```
Comment (sehz) [inline .github/workflows/ci.yml:596]: Still need this if SCCACHE is not being used?

**Review by sehz — CHANGES_REQUESTED** [review]

**Review by sehz — APPROVED** [review]
LGTM

**Pull request closed without merging**